src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorProvider.cs

```diff
@@ -95,39 +96,106 @@ public TagHelperDescriptorProvider(IEnumerable<TagHelperDescriptor> descriptors)
+                        else
+                        {
+                            matchingAttributes = attributes.Where(
+                                attribute => string.Equals(attribute.Key, attributeName, StringComparison.OrdinalIgnoreCase));
+                        }
+
+                        if (!matchingAttributes.Any())
                         {
                             return false;
                         }
+
+                        // At least one attribute where a name matches, now validate the value (if restricted).
+                        var equalsIndex = requiredAttribute.IndexOf('=');
+                        if (equalsIndex != -1)
+                        {
+                            var afterEquals = requiredAttribute[equalsIndex + 1];
+                            var valueOffset = 0;
+                            if (afterEquals == '\'' || afterEquals == '"')
+                            {
+                                valueOffset = 1;
+                            }
+                            var valueStart = equalsIndex + 1 + valueOffset;
+                            var valueLength = requiredAttribute.Length - equalsIndex - valueOffset * 2 - 2;
+                            var requiredValue = requiredAttribute.Substring(valueStart, valueLength);
+                            var selectorCharacter = requiredAttribute[equalsIndex - 1];
+
+                            switch (selectorCharacter)
+                            {
+                                case '^': // Value starts with
+                                    if (!matchingAttributes.Any(attribute => attribute.Value.StartsWith(requiredValue, StringComparison.Ordinal)))
+                                    {
+                                        return false;
+                                    }
+                                    break;
+                                case '$': // Value ends with
+                                    if (!matchingAttributes.Any(attribute => attribute.Value.EndsWith(requiredValue, StringComparison.Ordinal)))
+                                    {
+                                        return false;
+                                    }
+                                    break;
+                                default: // Value equals
+                                    if (!matchingAttributes.Any(attribute => string.Equals(attribute.Value, requiredValue, StringComparison.Ordinal)))
+                                    {
+                                        return false;
+                                    }
+                                    break;
+                            }
+                        }
                     }
 
                     return true;
                 });
         }
 
+        public static string GetRequiredAttributeName(string selector)
+        {
+            if (selector.Length <= 2 || selector[0] != '[')
```

Comment:
How confused will the system be if the "selector" (odd thing to call the original attribute name text) contains an `'='` but is not bracketed?